### PR TITLE
openstack-crowbar: List installed packages and diff between states

### PR DESCRIPTION
**Pull request by viccuad**
Hello,

This PR adds the collection of installed packages in each node both in plaintext and yaml, and diff results between states of cloud deployments (eg: clean deployment vs updated cloud differences).

It achieves it adding 2 ansible playbooks, list-installed-packages.yaml and diff-installed-packages.yml, as well as calls from the Crowbar pipeline to them.

Results of the playbooks are stored in `/root/rpms/*` & `/root/rpms/diffs/*` for all nodes (crowbar and cloud nodes). All of them are copied to the crowbar node, and archived as jenkins artifacts.

### CODE REVIEW

**Comment by viccuad** [comment]
I'm leaving on vacation until the 26h of May, so sadly this PR is a bit in an incomplete status. Please feel free to take, adapt, discard, etc.

The provided functionality should work, but:

TODO:
- Copying resulting files from the nodes to the crowbar node currently deadlocks; commented in code.
- For testing purposes, I am installing `iotop` package by hand in `list-installed-packages.yaml`, to be able to perform diffs of the deployments until updating deployments with MUs is in master.

**Comment by viccuad** [comment]
Running job with PR in: https://ci.suse.de/blue/organizations/jenkins/openstack-crowbar/detail/openstack-crowbar/317/pipeline

File: scripts/jenkins/ardana/ansible/list-installed-packages.yml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+            rpm -qa | sort > ~/rpms/rpms_{{ inventory_hostname }}_{{ state }}.txt
+          args:
+            warn: False
+            creates: "~/rpms/rpms_{{ inventory_hostname }}_{{ state }}.txt"
+
+        - name: Collect list of installed packages in yaml
+          shell: |
+            rpm -qa --qf "%{NAME}:\n  version: %{VERSION}\n  release: %{RELEASE}\n  disturl: %{DISTURL}\n" \
+            | perl -pe 'chomp;$_.=($.%4)?"\n":"\0"' | sort -z -t' ' -k1 | tr '\0' '\n' \
+            > ~/rpms/rpms_{{ inventory_hostname }}_{{ state }}.yaml
```
Comment (bmwiedemann) [inline scripts/jenkins/ardana/ansible/list-installed-packages.yml]: This is more complicated than it has to be.
You get the same result from
```bash
rpm -qa --qf "%{NAME}:\t  version: %{VERSION}\t  release: %{RELEASE}\t  disturl: %{DISTURL}\n" \
| sort | tr '\t' '\n' \
> ....yaml
```

And you can even drop the tr, if you go for a more JSON-like yaml:
```bash
rpm -qa --qf '"%{NAME}": \{ "version": "%{VERSION}", "release": "%{RELEASE}", "disturl": "%{DISTURL}" \}\n' |
 sort > ....yaml
```

It is still somewhat human-readable:
```yaml
"a2ps": { "version": "4.14", "release": "lp150.1.37", "disturl": "obs://build.opensuse.org/openSUSE:Leap:15.0/standard/9d231eacdd5ca84957a77f837758ec48-a2ps" }
"aaa_base-extras": { "version": "84.87+git20180409.04c9dae", "release": "lp150.8.1", "disturl": "obs://build.opensuse.org/openSUSE:Maintenance:9894/openSUSE_Leap_15.0_Update/43255c1ce72a835dd9154a2a1cec4a85-aaa_base.openSUSE_Leap_15.0_Update" }
```

**Comment by rsalevsky** [comment]
New PR will be created.

**Pull request closed without merging**